### DISCUSSION TITLE
Additional url parameters ?

### DISCUSSION THREAD

**Discussion by StarDustEins** [zauberzeug/nicegui#796] [Q&A] [answered]
### Question

I need to work with single sign on (SSO) , the main app calls my url like this (http://10.0.0.100:8080/home?access_token=xxx-xxx-xxx-xxx)

```py
@ui.page('/')
@ui.page('/{_:path}')
async def main():
    router = Router()
    @router.add('/')
    async def show_one(access_token: str = None):
        print(access_token)
        ui.label('Content One').classes('text-2xl')
ui.run()
```

here is my code , but print is None , how could i get the access_token ?

**Comment by StarDustEins** [comment]
my app is based on single_page_app demo

**Comment by rodja** [comment, marked as answer]
I think you mixed up a few things. This is working for me:

```py
from nicegui import ui

@ui.page('/')
@ui.page('/{_:path}')
async def main(access_token: Optional[str] = None):
    ui.label(access_token).classes('text-2xl')

ui.run()
```

Requesting with http://127.0.0.1:8080/?access_token=test shows a page with the text "test". If you are using the Router from single-page-app, the routes itself do not have any parameters.

**Comment by PierreFaraut** [comment]
Hello, 
Not sure to understand how does it work in the example of the single page app demo. 
For example here:
```python
@ui.page('/')  
@ui.page('/{_:path}')  
def main():
    router = Router()

    @router.add('/')
    def show_one():
        ui.label('Content One').classes('text-2xl')

    @router.add('/two')
    def show_two():
        ui.label('Content Two').classes('text-2xl')
```

Lets say that for the "show_two" function, I want to add a user id as a parameter. Like this: 
```python
def show_two(user_id: Optional[str] = None):
        ui.label(f'Content Two for user {user_id}').classes('text-2xl')
```
How would you do the parameter injection here? 

Thanks
Pierre

**Reply by rodja** [reply]
In the examples of `access_token` in the original question and your `user_id` I suspect you would be better of not injecting a parameter into every spa-page but rather in the initial main page builder as I showed in https://github.com/zauberzeug/nicegui/discussions/796#discussioncomment-5665628. If you really need parameters to be passed into your spa-page you will need to alter the implementation of the router class.
